autoscheduler/scraper/banner_requests.py

```diff
@@ -0,0 +1,187 @@
+                  str(int(time.time() * 1000)))
+    return session_id
+
+# Can semester be an enum?
+def get_term_code(year: str, semester: Semester, location: Location):
+    """ Generates a term code given a year and which semester
+
+        year: 4 character string, could/should probs be an integer
+
+        Semester: fall, spring, summer, wintermester?
+    """
+
+    if len(year) != 4:
+        raise ValueError("Year argument must be 4 characters long")
+
+    return year + str(semester.value) + str(location.value)
+
+class BannerRequests():
+    """ Handles basic banner requests """
+
+    def __init__(self, term: str):
+        base_url = 'compassxe-ssb.tamu.edu'
+
+        self.term_code = term
+
+        self.depts_url = ('https://%s/StudentRegistrationSsb/ssb/classSearch/get_subject?'
+                          'dataType=json&offset=1&term={term}&max={max}' % base_url)
+
+        self.course_search_url = ('https://%s/StudentRegistrationSsb/ssb/searchResults/'
+                                  'searchResults/?pageOffset=0&sortDirection=asc&'
+                                  'sortColumn=subjectDescription&txt_subject={subject}&'
+                                  'txt_term={term}&uniqueSessionId={uniqueSessionId}&'
+                                  'pageMaxSize={num_courses}' % base_url)
+
+        self.create_session_url = ('https://%s/StudentRegistrationSsb/ssb/term/search?'
+                                   'mode=search&dataType=json' % base_url)
+
+        self.reset_search_url = ('https://%s/StudentRegistrationSsb/ssb/classSearch/'
+                                 'resetDataForm') % base_url
+
+    async def create_session(self, session: ClientSession) -> str:
+        """ Begins the session and validates the session_id
+            Must be called in order to search for courses
+        """
+
+        session_id = generate_session_id()
+
+        data = {
+            'uniqueSessionId': session_id,
+            'term': self.term_code,
+        }
+
+        await session.post(self.create_session_url, data=data)
+
+        return session
+
+    async def get_courses(self, session: ClientSession, session_id: str, dept: str,
+                          amount: int) -> List[Dict]:
+        """ Retrieves all of the courses for a given department
+
+            dept: Department, a four letter string, such as CSCE
+            amount: max amount of courses to retrieve
+        """
+
+        data = {
+            'uniqueSessionId': session_id,
+            'term': self.term_code,
+            'subject': dept,
+            'num_courses': amount,
+        }
+
+        url = self.course_search_url.format(**data)
+
+        async with session.get(url) as response:
+            json = await response.json()
+
+        await self.reset_search(session)
+
+        data = json['data']
+
+        return data
+
+    async def get_departments(self, amount: int = 300) -> List[Dict]:

```

Comment:
```suggestion
    def get_departments(self, amount: int = 300) -> List[Dict]:
```
Should be made synchronous

Reply:
Yeah this was a mistake, `requests.get()` is a blocking call so there's no need for `async`. I originally had it there as I planned on using `ClientSession`, but it wasn't necessary. Fixing